eth/downloader/downloader.go

```diff
@@ -664,14 +664,6 @@ func (d *Downloader) findAncestor(p *peerConnection, remoteHeader *types.Header)
 		floor = int64(localHeight - MaxForkAncestry)
 	}
 
-	// We might have already downloaded this sidechain, if so we can abort early
-	if d.blockchain.HasBlock(remoteHeader.Hash(), remoteHeight) {
```

Comment:
You could also keep this if you added a check to ensure it's not doe in light sync:

```
if (d.mode == FullSync || d.mode == FastSync) && d.blockchain.HasBlock(remoteHeader.Hash(), remoteHeight) {
...
```

Reply:
The thing is, I'm not sure we would ever hit this, for two reasons:
1. An invalid sidechain with the current import logic would never save the remote head-block, 
2. The change in fetcher (or downloader?) to delay the last block(s) stops that block from reaching us anyway

Reply:
I remember that I was confused why we weren't hitting this clause in my tests, and it was some logic-clash with that delay thingy